krypto-lib/src/main/cpp/BitVector.h

```diff
@@ -444,7 +510,7 @@ class BitVector {
      */
     const BitVector<NUM_BITS> rightShift(unsigned int n){
         BitVector<NUM_BITS> result = BitVector<NUM_BITS>::zeroVector();
-        for(unsigned int i = n; i < NUM_BITS; ++i)
+        for(unsigned int i = NUM_BITS; i >= n; --i)
```

Comment:
I don't think this is used by any performance critical sections, but we should have a TODO to make this more efficient.